vllm/v1/core/scheduler.py

```diff
@@ -638,6 +657,7 @@ def finish_requests(
                     self.scheduled_req_ids.remove(request.request_id)
             else:
                 self.waiting.remove(request)
+                self._waiting_tokens -= self._get_request_total_tokens(request)
```

Comment:
If I'm understanding this properly, the number of waiting tokens is prompt + future decodes? Not just prompt tokens waiting to be scheduled?